numpy/core/arrayprint.py

```diff
@@ -521,6 +509,8 @@ def array2string(a, max_line_width=None, precision=None,
     if formatter is None:
         formatter = _formatter
 
+    if style is not None and a.shape == () and a.dtype.names is None:
+        return style(a[()])
```

Comment:
My concern is that that line doesn't fully solve the problem, not that it's worse than what we had before. My other concern is that that change isn't really in line with the title of the PR, but that's not too big a deal if they end up in separate commits after the squash